### PR TITLE
Add tags to Models

### PR DESCRIPTION
**Pull request by jean-lucas**
**note**: works on running the backend: https://github.com/scaleapi/scaleapi/pull/41353/files

Allow users to create tags on models

Example:

```python

# on existing models
m = client.models[0]
m.add_tag('some_tag')
m.add_tag(['a', 'b'])
print(m.tags)

# on model creation=
model = client.create_model(
    name="foo_model",
    reference_id="foo-model-ref",
    tags=["some tag"]
)
print(model.tags)

# delete

m.remove_tag(['a'])
```

Results can also be visualized on the models pages

### CODE REVIEW

**Comment by jean-lucas** [comment]
`add_tag` adn `remove_tag`

**Review by pfmark — APPROVED** [review]
minor comments, otherwise LGTM, nice work!

File: nucleus/model.py [hunk 1]
```diff
@@ -213,3 +221,58 @@ def run(self, dataset_id: str, slice_id: Optional[str]) -> str:
...
+            f"model/{self.id}/tag",
+            requests_command=requests.post,
+        )
+
+        if response.get("msg", False):
+            self.tags.extend([tags] if isinstance(tags, str) else tags)
+
+        return response
+
+    def remove_tag(self, tags: Union[str, List[str]]):
```
[the remark this answers is not in the document]
Comment (jean-lucas) [reply nucleus/model.py]: done 👍

File: tests/test_models.py [hunk 2]
```diff
@@ -115,3 +115,31 @@ def test_new_model_endpoints(CLIENT, dataset: Dataset):
...
+
+    model.add_tag("single tag")
+    model.add_tag(["tag_a", "tag_b"])
+
+    backend_model = testing_model(model_reference)
+    assert sorted(backend_model.tags) == sorted(
+        ["first_tag", "single tag", "tag_a", "tag_b"]
+    )
+
+    model.remove_tag("tag_a")
```
[the remark this answers is not in the document]
Comment (jean-lucas) [reply tests/test_models.py]: done

**Pull request merged**